models/user.go

```diff
@@ -0,0 +1,88 @@
+package models
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"golang.org/x/crypto/bcrypt"
+)
+
+type User struct {
+	ID        int64  `json:"id,omitempty"`
+	Name      string `json:"name,omitempty"`
+	Email     string `json:"email,omitempty"`
+	Password  string `json:"password,omitempty"`
+	AvatarURL string `json:"avatar_url,omitempty"`
+}
+
+const (
+	usersTable string = "users"
+)
+
+//Creates a user instance in the db and returns the user's id and the error
+func (db *DB) CreateUser(user *User) (int64, error) {
+	tx, err := db.Begin()
+	if err != nil {
+		return -1, err
+	}
+
+	user.Password, err = hashPassword(user.Password)
```

Comment:
I think we should be generating a salt and hashing the combined password+salt.

Reply:
From what I understand, the way `bcrypt` works is by generating the salt and hashing the password with it and combining them together, so I think this way is okay. This is what I'm going off of: https://auth0.com/blog/hashing-in-action-understanding-bcrypt/.

Reply:
Ya with bcrypt the salt is generated but we never need to worry ourselves with it.

Reply:
Ah, I see

Reply:
cool, I'm gonna resolve this and remove the DS_Store